### PR TITLE
Split artifacts per arch and re-enable updater for nightly builds

### LINKED ISSUE #3920: Split builds per os and per architecture to minimize artifacts size

### PR DESCRIPTION
**Pull request by pierremtb**
Fixes #3920

This allows for much smaller .zip files of build artifacts, making the process of downloading them for debugging or testing much easier (going from 500 to 100-200 MB).

Also noticed our nightly builds weren't being pushed correctly, fixing that here as well.

### CODE REVIEW

File: .github/workflows/build-test-publish-apps.yml [hunk 1]
```diff
@@ -63,12 +61,25 @@ jobs:
       - id: export_version
         run: echo "version=`cat package.json | jq -r '.version'`" >> "$GITHUB_OUTPUT"
 
+      - name: Prepare electron-builder.yml file for nightly
+        if: ${{ github.event_name == 'schedule' }}
+        run: |
+          yq -i '.publish[0].url = "https://dl.zoo.dev/releases/modeling-app/nightly"' electron-builder.yml
```
[the remark this answers is not in the document]
Comment (pierremtb) [reply .github/workflows/build-test-publish-apps.yml:67]: electron-updater has a better way to manage channels though, we should really migrate to that if we want to put it in the effort https://www.electron.build/tutorials/release-using-channels.html

**Comment by pierremtb** [comment]
image

Alright looking good build size wise. win arm and mac contain two artifacts for different reasons which is cranking up the size from 100ish to 200ish MB, but let's fix that in another PR

**Comment by pierremtb** [comment]
But it looks like we're still missing win arm actually in the end for publish. Left is a dummy ls from this branch, while right is the last good release
image

**Comment by pierremtb** [comment]
image

Alright, looking good now!

**Review by franknoirot — APPROVED** [review]
LGTM

**Pull request merged**